kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -258,28 +258,71 @@ private static class ExpressionEvalVisitor extends ExpressionVisitor<ColumnVecto
             this.input = input;
         }
 
+        /*
+        | Operand 1 | Operand 2 | `AND`      | `OR`       |
```

Comment:
nit: after seeing this in the code, now that the code is super clear and easy to follow, I guess we don't really need this. sry about that

Reply:
I honestly think it doesn't hurt and it clarifies the null semantics at least which aren't as obvious (I would at least want the rows with the null to be clarified somewhere)

Reply:
sgtm